### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by juhlig**
* External worker control has been dropped and moved directly into the pool module
* Rooster map has been replaced with an ETS table
* `prune/1` function to immediately shrink pool size back towards minimum

### CODE REVIEW

**Comment by Maria-12648430** [comment]
Looks good on a first glance. I'll have a more in-depth look tomorrow :)

File: src/hnc.erl [hunk 1]
```diff
@@ -52,7 +53,8 @@
 	idle:=non_neg_integer(),
 	out:=non_neg_integer(),
 	starting:=non_neg_integer(),
-	returning:=non_neg_integer()
+	returning:=non_neg_integer(),
+	stopping:=non_neg_integer()
```
Comment (Maria-12648430) [inline src/hnc.erl]: stopping is not returned from anywhere?

Comment (juhlig) [reply src/hnc.erl]: Ah, right... forgot to remove that ^^;

Comment (Maria-12648430) [reply src/hnc.erl]: Thought so ;)

**Comment by Maria-12648430** [comment]
Ok, I think this is good now :)

Btw, I just had the greatest idea =) Let's see what I think about it one hour from now, though ;)

**Comment by juhlig** [comment]
Ok, I'll merge it then, and wait to see what becomes of your greatest idea ;)

**Pull request merged**